### PR TITLE
Migration Update for Incorrect Answer List Handling

### PR DESCRIPTION
**Pull request by aaladaruncc**
This pull request introduces two new migration files aimed at updating the structure of incorrect answer lists in the database and merging previous migrations. The changes made are as follows:

## New Migration Files

1. **0054_auto_20241111_2121.py**:
   - **Functionality**: Updates the database schema for `QuestionMultipleChoice` model.
   - **Key Changes**:
     - **Data Transformation**: Implements a function to transform the `incorrect_answer_list` from an old format (list of strings) to a new format (list of dictionaries with answer and feedback).  
     - **Iteration**: Iterates through existing records and checks if the `incorrect_answer_list` is a list of strings, then modifies it accordingly.
     - **Data Saving**: Saves the updated data back to the database, ensuring that only the `incorrect_answer_list` field is updated.
   - **Dependencies**: This migration is dependent on the completion of `0053_remove_settings_timer_duration`.

2. **0055_merge_0054_auto_20241111_2121_0054_lecturesummary.py**:
   - **Purpose**: Merges two migration files into a single migration.
   - **Dependencies**: Merges dependencies from both `0054_auto_20241111_2121` and `0054_lecturesummary` migrations.
   - **Operations**: No operations are defined under this migration, functioning purely as a merge.

## Impact
- **Database Consistency**: Ensures that the structure of the `incorrect_answer_list` is consistent and up-to-date, facilitating easier data management and access.
- **Maintainability**: Helps future-proof the application by moving towards a more structured data format, which can accommodate additional features like feedback in the future.

### CODE REVIEW

**Comment by aaladaruncc** [comment]
The reason the Count Lines is failing because the only changes made were migration files.

**Comment by aaladaruncc** [comment]
The black test is failing due to a previous bug in Bruh's PR

File: api/models.py [hunk 1]
```diff
@@ -117,6 +117,25 @@ def to_json(self):
...
 
+    def to_json(self):
+        # Extract the list of answers from the new data structure
+        incorrect_answers = []
+        for item in self.incorrect_answer_list:
+            if isinstance(item, dict) and 'answer' in item:
+                incorrect_answers.append(item['answer'])
+            else:
+                # In case of any anomalies, include the item as is
+                incorrect_answers.append(item)
```
Comment (amhajja) [inline api/models.py:128]: Very nice; we need to add logging in the `else` body; I'll add a GitHub issue from this comment!

**Review by amhajja — APPROVED** [review]
Very nice!

**Pull request merged**